docs/source/notes/amp_examples.rst

```diff
@@ -0,0 +1,186 @@
+        grad_params = [p*(1./scaler.get_scale()) for p in grad_params]
+
+        # Compute the penalty term and add it to the loss
+        grad_norm = 0
+        for grad in grad_params:
+            grad_norm += grad.pow(2).sum()
+        grad_norm = grad_norm.sqrt()
+        loss = loss + grad_norm
+
+        # The usual scaling for backward will now accumulate leaf gradients that are appropriately scaled.
+        scaler.scale(loss).backward()
+        scaler.step(optimizer)
+        scaler.update()
+
+
+Gradient accumulation
+---------------------
+
+Gradient accumulation across iterations (between steps) can be implemented as follows::
+
+    scaler = AmpScaler()
+    ...
+    for i, (input, target) in enumerate(data):
+        output = model(input)
+        loss = loss_fn(output, target)
+        loss = loss/iters_to_accumulate
+        scaler.scale(loss).backward()
+        if (i + 1) % iters_to_accumulate == 0:
+            scaler.step(optimizer)
+            scaler.update()
+            optimizer.zero_grad()
+
+Switching gradient scaling on and off
+------------------------------------
+
+You can enable/disable gradient scaling globally (everywhere in the network a given :class:`AmpScaler` instance
+is used) by supplying a single ``enabled=True|False`` flag to that instance's constructor call::
+
+    scaler = AmpScaler(enabled=args.use_mixed_precision)
+    ...
+    for input, target in data:
+        optimizer.zero_grad()
+        output = model(input)
+        loss = loss_fn(output, target)
+        scaler.scale(loss).backward()
+        scaler.step(optimizer)
+        scaler.update()
+
+If ``enabled=False``, ``scaler.step(optimizer)`` is equivalent to ``optimizer.step()``, and
+the other methods (``scaler.scale``, ``scaler.update``) become no-ops.
+
+
+Working with Multiple Optimizers, Models, and Losses
+----------------------------------------------------
+
+A single :class:`AmpScaler` instance should scale all losses and step all optimizers.
+The usage is equivalent to prior single-optimizer examples.  The only additional constraint is
+that ``scaler.update()`` may only be called after all optimizers used this iteration have been stepped::
+
+    scaler = torch.cuda.amp.AmpScaler()
+    ...
+    for input, target in data:
+        optimizer0.zero_grad()
+        optimizer1.zero_grad()
+        output0 = model0(input)
+        output1 = model1(input)
+        loss0 = loss_fn(2 * output0 + 3 * output1, target)
+        loss1 = loss_fn(3 * output0 - 5 * output1, target)
+
+        scaler.scale(loss0).backward(retain_graph=True)
+        scaler.scale(loss1).backward()
+
+        # You can choose which optimizers receive explicit unscaling
+        scaler.unscale(optimizer0)
+
+        scaler.step(optimizer0)
+        scaler.step(optimizer1)
+        scaler.update()
+
+Note that the decision to invoke an explicit ``unscale`` can be made independently for each optimizer.
+
+.. warning::
```

Comment:
It seems like this note should be in the section about "Working with Unscaled Gradients," right? The prohibition applies universally.